startUp.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+
+import sys
+import subprocess
+import time
+#process names from text file stored into array
+#sleep after each check for 2 secs
+
+
+def main(): 
+    #read program names from file and put them into an array
+    fileName = open(sys.argv[1])
+    programs = fileName.read().splitlines()
+    fileName.close()
+    
+    numberOfPrograms = programs.__len__()
+    MAX_RESTART = 10
+    timesRestarted = [0] * numberOfPrograms
```

Comment:
I would actually recommend keeping a dict for this, where the key is the path to the program and the value is the number of times restarted.